src/errors.ts

```diff
@@ -0,0 +1,6 @@
+export class AuthenticationError extends Error {
+  constructor(name: string, message: string) {
+    super(message);
+    this.name = name;
```

Comment:
I think I rather have both `error` and `error_description` in the error.

```
super(error_description);
this.error = error;
this.error_description = error_description;
```

Reply:
Do you recommend not extending the Error class?  I've mapped `error_description` to `message` and `error` to the name.

Reply:
We should absolutely extend the error class, but we should keep the original properties (`error` and `error_description`) intact so the dev can do:

```
catch ({error, error_description}) {
  //handle it
}
```

Reply:
Also, if there's more information in the error that we receive (like `state` or something else), they should be here as well.